dockerfiles/Dockerfile.router

```diff
@@ -1,15 +1,25 @@
 FROM debian:bullseye-slim
 
 ARG ROUTER_RELEASE=latest
+ARG DEBUG_IMAGE=false
 
 WORKDIR /dist
 
 # Install curl
 RUN \
   apt-get update -y \
   && apt-get install -y \
-    curl \
-  && rm -rf /var/lib/apt/lists/*
+    curl
+
+# If debug image, install heaptrack and make a data directory
+RUN \
+  if [ "${DEBUG_IMAGE}" = "true" ]; then \
+    apt-get install -y heaptrack && \
+    mkdir data; \
+  fi
+
+# Clean up apt lists
+RUN rm -rf /var/lib/apt/lists/*
 
 # Run the Router downloader which puts Router into current working directory
 RUN curl -sSL https://router.apollo.dev/download/nix/${ROUTER_RELEASE}/ | sh
```

Comment:
the router binary that is used here will be without debug symbols, right? Will heaptrack be able to display enough information?

Reply:
You get the output we were looking at this morning. Things are mangled but you can roughly work out what is happening. I do think we can improve this further down the line, this is a step along the way.